Source/DafnyLanguageServer.Test/Various/CounterExampleTest.cs

```diff
@@ -1003,61 +1003,74 @@ class Cla__ss {
     }
 
     [TestMethod]
-    public void DafnyModelTypeMultipleArguments() {
-      var type = DafnyModelType.FromString("seq<int, char>");
-      Assert.AreEqual("seq", type.Name);
-      Assert.AreEqual(2, type.TypeArgs.Count);
-      Assert.AreEqual("int", type.TypeArgs[0].Name);
-      Assert.AreEqual("char", type.TypeArgs[1].Name);
-      Assert.AreEqual(0, type.TypeArgs[0].TypeArgs.Count);
-      Assert.AreEqual(0, type.TypeArgs[1].TypeArgs.Count);
-    }
+    public async Task UnboundedIntegers() {
+      var source = @"
+      ghost const NAT64_MAX := 0x7fff_ffff_ffff_ffff
+
+      newtype nat64 = x | 0 <= x <= NAT64_MAX
 
+      function method plus(a: nat64, b: nat64): nat64 {
+        a + b
+      }".TrimStart();
+      var documentItem = CreateTestDocument(source);
+      await client.OpenDocumentAndWaitAsync(documentItem, CancellationToken);
+      var counterExamples = (await RequestCounterExamples(documentItem.Uri)).ToArray();
+      Assert.AreEqual(1, counterExamples.Length);
+      Assert.IsTrue(counterExamples[0].Variables.ContainsKey("a:int"));
+      Assert.IsTrue(counterExamples[0].Variables.ContainsKey("b:int"));
+      var a = long.Parse(counterExamples[0].Variables["a:int"]);
+      var b = long.Parse(counterExamples[0].Variables["b:int"]);
+      Assert.IsTrue(a + b < a || a + b < b);
+    }
 
     [TestMethod]
-    public void DafnyModelTypeNestedArguments() {
-      var type = DafnyModelType.FromString("seq<map<char, int>>");
-      Assert.AreEqual("seq", type.Name);
-      Assert.AreEqual(1, type.TypeArgs.Count);
-      Assert.AreEqual("map", type.TypeArgs[0].Name);
-      Assert.AreEqual(2, type.TypeArgs[0].TypeArgs.Count);
-      Assert.AreEqual("char", type.TypeArgs[0].TypeArgs[0].Name);
-      Assert.AreEqual("int", type.TypeArgs[0].TypeArgs[1].Name);
-      Assert.AreEqual(0, type.TypeArgs[0].TypeArgs[0].TypeArgs.Count);
-      Assert.AreEqual(0, type.TypeArgs[0].TypeArgs[1].TypeArgs.Count);
+    public async Task DatatypeWithPredicate() {
+      var source = $@"
+      module M {{
+        datatype D = C(i:int) {{ 
+          predicate p() {{true}}
+        }}
+
+        method test(d: D) {{
+          if (d.p()) {{
+            assert d.i != 123;
+          }}
+        }}
+      }}".TrimStart();
```

Comment:
```suggestion
      var source = @"
      module M {
        datatype D = C(i:int) {
          predicate p() {true}
        }

        method test(d: D) {
          if (d.p()) {
            assert d.i != 123;
          }
        }
      }".TrimStart();
```
Can you please remove the interpolant "$" on the other tests where it's not necessary? That will improve legibility.